lib/chef/resource/chef_client_launchd.rb

```diff
@@ -110,21 +110,38 @@ class ChefClientLaunchd < Chef::Resource
           end
         end
 
-        launchd "com.chef.chef-client" do
+        launchd "com.#{Chef::Dist::SHORT}.#{Chef::Dist::CLIENT}" do
           username new_resource.user
           working_directory new_resource.working_directory
           start_interval new_resource.interval * 60
           program_arguments ["/bin/bash", "-c", client_command]
           environment_variables new_resource.environment unless new_resource.environment.empty?
           nice new_resource.nice
           low_priority_io true
+          notifies :sleep, "chef_sleep[Sleep before client restart]", :immediately
           action :enable
         end
+
+        launchd "com.#{Chef::Dist::SHORT}.restarter" do
+          username "root"
+          watch_paths ["/Library/LaunchDaemons/com.#{Chef::Dist::SHORT}.#{Chef::Dist::CLIENT}.plist"]
+          standard_out_path "#{::File.join(new_resource.log_directory, new_resource.log_file_name)}"
+          standard_error_path "#{::File.join(new_resource.log_directory, new_resource.log_file_name)}"
+          program_arguments ["/bin/bash",
+                             "-c",
+                             "echo; echo #{Chef::Dist::PRODUCT} launchd daemon config has been updated. Manually unloading and reloading the daemon; echo Now unloading the daemon; launchctl unload /Library/LaunchDaemons/com.#{Chef::Dist::SHORT}.#{Chef::Dist::CLIENT}.plist; sleep 2; echo Now loading the daemon; launchctl load /Library/LaunchDaemons/com.#{Chef::Dist::SHORT}.#{Chef::Dist::CLIENT}.plist"]
+          action :enable
+        end
+
+        chef_sleep "Sleep before client restart" do
```

Comment:
this could probably use a comment because it isn't clear at all why this sleep is in here.

Reply:
I half typed it and then deleted it. Clearly I should have kept going